stdlib/streams/src/main/ballerina/streams/windows.bal

```diff
@@ -1794,3 +1794,158 @@ public function timeAccum(any[] windowParameters, function (StreamEvent[])? next
     TimeAccumulatingWindow timeAccumulatingWindow1 = new(nextProcessPointer, windowParameters);
     return timeAccumulatingWindow1;
 }
+
+public type HoppingWindow object {
+    public int timeInMilliSeconds;
+    public int hopeTime;
+    public any[] windowParameters;
+    public LinkedList currentEventQueue;
+    public StreamEvent? resetEvent;
+    public task:Timer? timer;
+    public boolean isStart;
+    public function (StreamEvent[])? nextProcessPointer;
+
+    public function __init(function (StreamEvent[])? nextProcessPointer, any[] windowParameters) {
+        self.nextProcessPointer = nextProcessPointer;
+        self.windowParameters = windowParameters;
+        self.timeInMilliSeconds = 0;
+        self.hopeTime = 0;
+        self.resetEvent = ();
+        self.timer = ();
+        self.currentEventQueue = new();
+        self.isStart = true;
+        self.initParameters(self.windowParameters);
+    }
+
+    public function initParameters(any[] parameters) {
+        if (parameters.length() == 2) {
+            any parameter0 = parameters[0];
+            if (parameter0 is int) {
+                self.timeInMilliSeconds = parameter0;
+            } else {
+                error err = error("Hopping window's first parameter, windowTime should be of type int");
+                panic err;
+            }
+
+            any parameter1 = parameters[1];
```

Comment:
give a proper name for the attributes. Let's not name it as parameter0, parameter1 and etc..

Reply:
Done